workalendar_holidays/wizards/workalendar_holiday_import.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+############################################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2011 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: Jose Suniaga <josemiguel@vauxoo.com>
+#    planned by: Jose Suniaga <josemiguel@vauxoo.com>
+############################################################################
+
+import logging
+
+from dateutil.relativedelta import relativedelta
+from openerp import _, api, fields, models
+from openerp.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from workalendar import africa
+    from workalendar import asia
+    from workalendar import america
+    from workalendar import canada
+    from workalendar import europe
+    from workalendar import oceania
+    from workalendar import usa
+except ImportError:
+    _logger.info('Cannot import workalendar')
+
+_INTERVALS = {
+    'days': lambda interval: relativedelta(days=interval),
+    'weeks': lambda interval: relativedelta(days=7*interval),
+    'months': lambda interval: relativedelta(months=interval),
+    'years': lambda interval: relativedelta(months=12*interval),
+}
+
+
+class WorkalendarHolidayImport(models.TransientModel):
+
+    _name = 'wizard.workalendar.holiday.import'
+
+    @api.multi
+    @api.depends('start_date', 'interval_number', 'interval_type')
+    def _compute_end_date(self):
+        for wiz in self:
+            wiz.end_date = fields.Date.from_string(wiz.start_date) + \
+                _INTERVALS[wiz.interval_type](wiz.interval_number)
+
+    start_date = fields.Date(
+        string='Start Date', default=fields.Date.today, readonly=1)
+    end_date = fields.Date(
+        string='End Date', compute='_compute_end_date')
+    interval_number = fields.Integer('Interval', default=1)
+    interval_type = fields.Selection([
+        ('days', 'Day(s)'),
+        ('weeks', 'Week(s)'),
+        ('months', 'Month(s)'),
+        ('years', 'Year(s)')],
+        string='Type', default='days', required=True)
+    calendar_id = fields.Many2one(
+        'resource.calendar', string="Work Time")
+    country_group_id = fields.Many2one(
+        'res.country.group', string="Continent", required=True)
+    country_id = fields.Many2one(
+        'res.country', string="Country", required=True)
+
+    @api.model
+    def _get_workalendar_module(self, continent_name, country_id=False):
+        country_libs = {
+            self.env.ref('base.ca').id: canada,
+            self.env.ref('base.us').id: usa
+        }
+        if country_id in country_libs:
+            return country_libs[country_id]
+        name = str(continent_name).lower()
+        libs = [africa, america, asia, europe, oceania]
+        module = filter(lambda x: (x.__name__).split('.')[1] == name, libs)
```

Comment:
hi @josemoralesp thanks for the feedback, in this case after comprehension list i think that 'any' function is not useful because return a boolean, and i need the matched value from tuple

_return matched value from 'libs' tuple_
```
module = [lib for lib in libs if (lib.__name__).split('.')[1] == name]
return module and module[0] or None
```

instead:

_return a boolean_
```
return any([lib for lib in libs if (lib.__name__).split('.')[1] == name])
```